asserts/snapasserts/validation_sets.go

```diff
@@ -659,6 +659,12 @@ func (v *ValidationSets) RequiredSnaps() []string {
 	return names
 }
 
+// SnapConstrained returns true if the given snap is constrained by any of the
+// validation sets known to this type.
```

Comment:
I see it was used before but not anywhere else in the code base, but the expression "this type" is a bit strange to me in this context.  

Given the original doc comment for the type, I would try to replace  "validation sets known to this type" with "validation sets part of this combination" or "validation sets known to this type" with "validation sets part of this ValidationSets". Please do the same for the other methods with the same kind of doc comment formulation.